starky/src/verifier.rs

```diff
@@ -164,19 +164,22 @@ where
     // where the "real" quotient polynomial is `t(X) = t_0(X) + t_1(X)*X^n + t_2(X)*X^{2n} + ...`.
     // So to reconstruct `t(zeta)` we can compute `reduce_with_powers(chunk, zeta^n)` for each
     // `quotient_degree_factor`-sized chunk of the original evaluations.
-    for (i, chunk) in quotient_polys
-        .chunks(stark.quotient_degree_factor())
-        .enumerate()
-    {
-        ensure!(
-            vanishing_polys_zeta[i] == z_h_zeta * reduce_with_powers(chunk, zeta_pow_deg),
-            "Mismatch between evaluation and opening of quotient polynomial"
-        );
+    // TODO: this should be done only with the iterator
```

Comment:
I'm not sure what you mean by this

Reply:
That it should be a way of avoiding the if let converting the option quotient_polys to iter. But I don't remember why it didn't worked when I tried.